streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateUpdateTask.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.internals;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.TopicPartition;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Updates the state for all Global State Stores.
+ */
+public class GlobalStateUpdateTask implements GlobalStateMaintainer {
+
+    private class SourceNodeAndDeserializer {
+        private final SourceNode sourceNode;
+        private final RecordDeserializer deserializer;
+
+        SourceNodeAndDeserializer(final SourceNode sourceNode,
+                                  final RecordDeserializer deserializer) {
+            this.sourceNode = sourceNode;
+            this.deserializer = deserializer;
+        }
+    }
+
+    private final ProcessorTopology topology;
+    private final InternalProcessorContext processorContext;
+    private final Map<TopicPartition, Long> offsets = new HashMap<>();
+    private final Map<String, SourceNodeAndDeserializer> deserializers = new HashMap<>();
+    private final GlobalStateManager stateMgr;
+
+
+    public GlobalStateUpdateTask(final ProcessorTopology topology,
```

Comment:
My gut feeling is that we do not need to mimic a `ProcessorTopology` and `InternalProcessorContext` interface for this task, as it is a very special task whose topology will just be a list of source topics and a list of state stores, making them as generic interfaces would just introduce one-time classes like `GlobalProcessorContext` in which lots of its functions will not be required at all. Instead we can just e.g. pass into it a list of source topics, a map between topics to state stores, etc and let it run its only loop for fetching + updating stores, etc.